src/services/worker/tasks/burn-rates.task.ts

```diff
@@ -26,294 +11,150 @@ export class BurnRatesTask {
+      const clusterError =
+        aggrs[index].burnRate.error || aggrs[index].balance.error;
+      if (
+        (clusterError &&
+          SolidityErrors.isError(clusterError, 'ClusterIsLiquidated')) ||
+        aggrs[index].isLiquidated.value
+      ) {
+        await this._clusterService.update(
+          {
+            owner: record.owner,
+            operatorIds: record.operatorIds,
           },
-          async (result: AtomicRetryFailedResult) => {
-            throw new Error(
-              `Web3Provider.currentBlockNumber failed: ${JSON.stringify({
-                result,
-                record,
-              })}`,
-            );
+          {
+            balance: null,
+            burnRate: null,
+            isLiquidated: true,
+            liquidationBlockNumber: null,
           },
         );
+        continue;
+      }
 
-        // Prepare final data
-        record.cluster = JSON.stringify(record.cluster);
-        // If cluster won't be liquidated, at which block number his balance becomes zero
-        // check if burnRate and balance are valid, > 0
-        if (record.burnRate > 0 && record.balance > 0) {
-          // Calculate in which block the cluster will start to be ready for liquidation
-          const liveUntilBlockByBalance =
+      const burnRate = aggrs[index].burnRate.value;
+      const balance = aggrs[index].balance.value;
+      const isLiquidated = aggrs[index].isLiquidated.value;
+      // If cluster won't be liquidated, at which block number his balance becomes zero
+      // check if burnRate and balance are valid, > 0
+      if (!Number.isNaN(burnRate) && !Number.isNaN(balance)) {
+        record.balance = balance;
+        record.burnRate = burnRate;
+        record.isLiquidated = isLiquidated;
+        // Calculate in which block the cluster will start to be ready for liquidation
+        const liveUntilBlockByBalance =
+          currentBlockNumber +
+          record.balance / record.burnRate -
+          minimumBlocksBeforeLiquidation;
+
+        // Calculate balance at the block when the cluster will start to be ready for liquidation
+        const latestLiveBlockBalance =
+          record.balance -
+          (liveUntilBlockByBalance - currentBlockNumber) * record.burnRate;
+
+        // Get minimum collateral amount (in SSV)
+        const collateralAmount =
+          +(await Web3Provider.getMinimumLiquidationCollateral());
+
+        if (record.balance <= collateralAmount) {
+          // If balance less than minimum collateral amount set a flag to liquidate asap
+          record.liquidationBlockNumber = currentBlockNumber;
+        } else if (latestLiveBlockBalance < collateralAmount) {
+          // If balance at the block when the cluster will start to be ready for liquidation
+          // less than minimum collateral amount, set the block when need to liquidate that to get at least minimum collateral amount
+          record.liquidationBlockNumber =
```

Comment:
This method looks too big, de-composite it to smaller methods corresponding for specific logical parts